filecache.go

```diff
@@ -16,54 +18,171 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+const (
+	DownloadMangerS3 = iota
+	DownloadMangerDropbox
+)
+
+var (
+	errInvalidURLPath = errors.New("invalid URL path")
+)
+
+type DownloadManager int
+
+// DownloadRecord contains information about a file which will be downloaded
+type DownloadRecord struct {
+	Manager DownloadManager
+	Path    string
+	Headers map[string]string
+}
+
+type recordDownloader = func(downloadRecord *DownloadRecord, localFile *os.File) error
```

Comment:
This should be public because the idea here was that people can drop in a different function.